medallion/backends/mongodb_backend.py

```diff
@@ -112,9 +117,14 @@ def add_objects(self, api_root, id_, objs, request_time):
         failed = 0
         succeeded = 0
         for new_obj in objs["objects"]:
-            existing_entry = objects.find_one({"collection_id": id_,
-                                               "id": new_obj["id"],
-                                               "modified": new_obj["modified"]})
+            if "modified" in new_obj:
```

Comment:
Can we refactor this to remove duplication? create a dictionary with `collection` and `id`, and then conditionally add `modified` if `new_obj` has that key.

Reply:
Unrelated, it might be good to change to function argument `id_` to be `collection_id`.

Reply:
something like this?

```python
mongo_query = {"collection_id": collection_id, "id": new_obj["id"]}
if "modified" in new_obj:
   mongo_query["modified"] = new_obj["modified"]
existing_entry = objects.find_one(mongo_query)
```